src/lib.rs

```diff
@@ -1542,6 +1542,21 @@ impl Build {
             "Expand may only be called for a single file"
         );
 
+        let is_asm = self
+            .files
+            .iter()
+            .map(std::ops::Deref::deref)
+            .find_map(AsmFileExt::from_path)
+            .is_some();
+
+        if compiler.family == (ToolFamily::Msvc { clang_cl: true }) && !is_asm {
```

Comment:
I wasn't 100% sure its needed. But since `!is_asm` is added in PR #747 (continuation of #514?), so i added it here too.

Reply:
#747 uses `is_assembler_msvc` instead of just. `is_assembler` though:


```rust
let msvc = target.contains("msvc");
let is_assembler_msvc = msvc && asm_ext == Some(AsmFileExt::DotAsm);
```

So it will add `--` as long as the target is not msvc, not sure which version is better.

IMO if `compiler.family` is msvc `clang-ci` then it's probably a MSVC target.